builder/vsphere/driver/datastore_test.go

```diff
@@ -36,19 +36,35 @@ func TestDatastoreIsoPath(t *testing.T) {
 			isoPath: "[datastore][] /dir/subdir/file",
 			valid:   false,
 		},
+		{
+			isoPath:  "   [datastore] /dir/subdir/file",
+			filePath: "/dir/subdir/file",
+			valid:    true,
+		},
+		{
+			isoPath:  "[datastore] /dir/subdir/file     ",
+			filePath: "/dir/subdir/file",
+			valid:    true,
+		},
+		// Test case for #9846
+		{
+			isoPath:  "[ISO-StorageLun9] Linux/rhel-8.0-x86_64-dvd.iso",
+			filePath: "Linux/rhel-8.0-x86_64-dvd.iso",
+			valid:    true,
+		},
```

Comment:
Yay, more tests; I think we should add tests for paths with spaces in there too, if they can happen ?

Reply:
done!! ✅